### PR TITLE
[FIX] Rearrange parameters of analyze_cigar()

### LINKED ISSUE #52: [BUG] Mixed up output string.
This bug came out of an discussion of: https://github.com/seqan/iGenVar/pull/49#discussion_r571115836

eldariont:
`test/api/junction_detection_test.cpp` lines +17 to +20:
```cpp
// Reference\tm2257/8161/CCS\t41972616\tForward\tRead \t0\t2294\tForward\tchr21
// INS from Primary Read - Sequence Type: Reference; Sequence Name: m2257/8161/CCS; Position: 41972616; Orientation: Reverse
//                         Sequence Type: Read; Sequence Name: 0; Position: 3975; Orientation: Reverse
//                         Chromosome: chr21
```
> I'm confused by this string:
> `// Reference\tm2257/8161/CCS\t41972616\tForward\tRead \t0\t2294\tForward\tchr21`
> It should consist of breakend1, breakend2 and the read name it was detected from. However, the read name here is chr21 and the reference chromosome of breakend1 is m2257/8161/CCS\t41972616. Those are swapped but I don't immediately see why 😕

-----
These output is coming from:
`iGenVar/src/detect_breakends/junction_detection.cpp`
and are built in:
`iGenVar/include/junction.hpp` lines 41 to 46:
```cpp
template <typename stream_t>
inline stream_t operator<<(stream_t && stream, junction const & junc)
{
    stream << junc.get_mate1() << '\t' << junc.get_mate2() << '\t' << junc.get_read_name();
    return stream;
}
```

`iGenVar/include/breakend.hpp` lines 35 to 43:
```cpp
template <typename stream_t>
inline stream_t operator<<(stream_t && stream, breakend const & b)
{
    stream << ((b.seq_type == sequence_type::reference) ? "Reference" : "Read ") << '\t'
           << b.seq_name << '\t'
           << b.position  << '\t'
           << ((b.orientation == strand::forward) ? "Forward" : "Reverse");
    return stream;
}

Probably there is a mixup in the passing of chromosome and read in `retrieve_aligned_segments` or `analyze_cigar`.

### PR DESCRIPTION
**Pull request by Irallia**
Resolves #52 

I made a mix up in the analyze_cigar(), which caused a wrong output.
=> I Rearranged the parameters of analyze_cigar()
I also made const parameter const, repaired the tests and updated the documentation

### CODE REVIEW

**Review by eldariont — APPROVED** [review]
Thanks @Irallia for finding and fixing the problem! :+1:

File: test/api/junction_detection_test.cpp [hunk 1]
```diff
@@ -15,22 +15,22 @@
 // }
 
 // Explanation for the stings:
```
Comment (SGSSGene) [inline test/api/junction_detection_test.cpp:17]: stings → strings

File: src/detect_breakends/junction_detection.cpp [hunk 2]
```diff
@@ -266,17 +266,17 @@ void detect_junctions_in_alignment_file(const std::filesystem::path & alignment_
...
-        std::string ref_name = ref_ids[ref_id];
+        const std::string query_name            = seqan3::get<seqan3::field::id>(rec);                      // 1: QNAME
+        const seqan3::sam_flag flag             = seqan3::get<seqan3::field::flag>(rec);                    // 2: FLAG
+        const int32_t ref_id                    = seqan3::get<seqan3::field::ref_id>(rec).value_or(0);      // 3: RNAME
+        const int32_t pos                       = seqan3::get<seqan3::field::ref_offset>(rec).value_or(0);  // 4: POS
+        const uint8_t mapq                      = seqan3::get<seqan3::field::mapq>(rec);                    // 5: MAPQ
+        std::vector<seqan3::cigar> cigar        = seqan3::get<seqan3::field::cigar>(rec);                   // 6: CIGAR
+        const auto seq                          = seqan3::get<seqan3::field::seq>(rec);                     // 10:SEQ
+        auto tags                               = seqan3::get<seqan3::field::tags>(rec);
+        const auto header_ptr                   = seqan3::get<seqan3::field::header_ptr>(rec);
```
Comment (SGSSGene) [inline src/detect_breakends/junction_detection.cpp:277]: uh! very nice to read assignment table! :+1:

**Review by SGSSGene — CHANGES_REQUESTED** [review]
Besides a minor typo (that was not introduced by your changes).  It looks good to me.

**Review by SGSSGene — APPROVED** [review]
:+1:

**Pull request merged**